distributed/worker.py

```diff
@@ -3459,9 +3595,11 @@ def meets_resource_constraints(self, key: str) -> bool:
 
         return True
 
-    async def _maybe_deserialize_task(self, ts, *, stimulus_id):
-        if not isinstance(ts.run_spec, SerializedTask):
-            return ts.run_spec
+    async def _maybe_deserialize_task(
+        self, ts: TaskState, *, stimulus_id: str
+    ) -> tuple[Callable, tuple, dict[str, Any]] | None:
+        if ts.run_spec is None:
+            return None
```

Comment:
This appears to be the only logical change in here. I checked the code again and this is safe right now. 

There is an optimization possible where we store the deserialized runspec on this attribute (function, args, kwargs, see below) in case the task needs to be recomputed and is not forgotten in between. We're not doing this right now, maybe we did in the past. For the current state of the code this is fine